lib/etcd_lib.py

```diff
@@ -38,18 +38,24 @@ def get_ingress_addresses(endpoint_name):
 
 
 def get_ingress_address(endpoint_name):
-    """Returns an ingress-address belonging to the named endpoint, if
-    available. Falls back to private-address if necessary."""
-    return get_ingress_addresses(endpoint_name)[0]
+    """Returns an ingress-address belonging to the named endpoint.
+
+    * Sort addresses by ipv4 first, then ipv6 next
+    * Falls back to private-address if necessary.
+    """
+    all_addrs = get_ingress_addresses(endpoint_name)
+    return sorted(all_addrs, key=lambda i: ip_address(i).version)[0]
```

Comment:
sort the addresses in such a way that ipv4 pops to the top first:

notice that python `sorted` doesn't alter any other order  (ie -- it doesn't deep sort)

```python
>>> i = [1,4,"a",3,"b",5]
>>> sorted(i, key=lambda k: type(k).__name__)
[1, 4, 3, 5, 'a', 'b']
```